### PR TITLE
Add response gzip,brotli compression

### PR DESCRIPTION
**Pull request by qiwzhang**
Add a flag --enable_response_compression to enable gzip and brotli compression for response body.

### CODE REVIEW

**Review by nareddyt — APPROVED** [review]

**Comment by qiwzhang** [comment]
https://github.com/GoogleCloudPlatform/oss-test-infra/pull/1555

File: tests/integration_test/compression_test/compression_test.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+		{
+			desc:       "accept-encode is gzip,br. choose gzip",
+			httpMethod: "GET",
+			method:     "/v1/shelves/100/books?key=api-key",
+			headers:    http.Header{"accept-encoding": []string{"gzip,br"}},
+			wantResp:   `{"books":[{"id":"1001","title":"Alphabet"}]}`,
+			wantDecode: "gzip",
+		},
+		{
+			desc:       "accept-encode is gzip;q=0.1,br;q=0.9. choose br",
```
Comment (TAOXUY) [inline tests/integration_test/compression_test/compression_test.go:86]: "accept-encode is gzip" => "accept-encode are gzip and br with weights"?

File: tests/integration_test/compression_test/compression_test.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+			httpMethod: "GET",
+			method:     "/v1/shelves/100/books?key=api-key",
+			headers:    http.Header{"accept-encoding": []string{"gzip;q=0.1,br;q=0.9"}},
+			wantResp:   `{"books":[{"id":"1001","title":"Alphabet"}]}`,
+			wantDecode: "br",
+		},
+		{
+			desc:       "accept-encode is gzip;q=0.9,br;q=0.1. choose gzip",
```
Comment (TAOXUY) [inline tests/integration_test/compression_test/compression_test.go:94]: ditto.

**Review by TAOXUY — APPROVED** [review]

File: tests/integration_test/compression_test/compression_test.go (hunk 1 above)
Comment (qiwzhang) [reply tests/integration_test/compression_test/compression_test.go]: Changed to "accept-encode is [gzip;q=0.1,br;q=0.9]".  use the actual header in the description

File: tests/integration_test/compression_test/compression_test.go (hunk 2 above)
Comment (qiwzhang) [reply tests/integration_test/compression_test/compression_test.go]: Done

**Comment by dorsec** [comment]
Can this also work for Protobuf responses?
Seems like it only supports JSON-transcoded responses

**Comment by qiwzhang** [comment]
For now, it doesn't support protobuf response.  It only compress with default content-types as:
```
application/javascript, application/json, application/xhtml+xml, image/svg+xml, text/css, text/html, text/plain, text/xml.
```

As documented [here](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/compressor_filter)

**Pull request merged**